packages/legend/src/Legend.tsx

```diff
@@ -0,0 +1,805 @@
+      featureSelectionLabelRawStr = featureSelectionLabelRaw?.at(-1);
+    } else if (typeof featureAggregationStrategy === 'number') {
+      const i = featureAggregationStrategy;
+      featureSelectionLabelRawStr = featureSelectionLabelRaw?.[i];
+    } else if (featureAggregationStrategy === 'sum') {
+      featureSelectionLabelRawStr = 'Sum of features';
+    } else if (featureAggregationStrategy === 'mean') {
+      featureSelectionLabelRawStr = 'Mean of features';
+    } else {
+      // Default to the first feature.
+      // featureAggregationStrategy was null.
+      featureSelectionLabelRawStr = featureSelectionLabelRaw?.[0];
+    }
+    const combinedMissing = combineMissings(missing ?? null, featureAggregationStrategy ?? null);
+    const featureSelectionLabel = combinedMissing
+      ? `${featureSelectionLabelRawStr} (${Math.round(combinedMissing * 100)}% NaN)`
+      : featureSelectionLabelRawStr;
+
+    // Include obsType in the label text (perhaps only when multi-obsType).
+    const obsLabel = capitalize(obsType ?? null);
+
+    // If the parent component wants to consider selections, then
+    // use the selected feature for the label. Otherwise,
+    // show the feature type.
+    const featureLabel = considerSelections
+      ? (featureSelectionLabel || capitalize(featureValueType ?? null))
+      : capitalize(featureValueType ?? null);
+
+    const mainLabel = showObsLabel ? obsLabel : featureLabel;
+    const subLabel = showObsLabel ? featureLabel : null;
+    const hasSubLabel = subLabel !== null;
+
+    if (!isSetColor) {
+      g
+        .append('text')
+        .attr('text-anchor', hasSubLabel ? 'start' : 'end')
+        .attr('dominant-baseline', 'hanging')
+        .attr('x', hasSubLabel ? 0 : width - 4)
+        .attr('y', 0)
+        .text(mainLabel ?? '')
+        .style('font-size', '10px')
+        .style('fill', foregroundColor);
+
+      if (hasSubLabel) {
+        g
+          .append('text')
+          .attr('text-anchor', 'end')
+          .attr('dominant-baseline', 'hanging')
+          .attr('x', width)
+          .attr('y', titleHeight)
+          .text(subLabel ?? '')
+          .style('font-size', '9px')
+          .style('fill', foregroundColor);
+      }
+    }
+  }, [width, height, featureValueColormap, featureValueColormapRange, localRange,
+    considerSelections, obsType, obsColorEncoding, featureSelection,
+    isDarkTheme, featureValueType, extent, featureLabelsMap,
+    spatialChannelColor, obsSetColor, obsSetSelection, isSetColor, theme,
+    contourPercentiles, contourThresholds, contoursFilled, contoursVisible,
+    pointsVisible, featureAggregationStrategy, setFeatureValueColormapRange,
+    dynamicHeight, highContrast, isStaticColor, missing, showObsLabel, staticColor,
+  ]);
+
+  // Determine if interactive slider should be shown
+  const showInteractiveSlider = (
+    setFeatureValueColormapRange
+    && ['geneSelection', 'geneExpression'].includes(obsColorEncoding ?? '')
```

Comment:
```suggestion
    && obsColorEncoding == 'geneSelection'
```

Reply:
Ah my bad, I see what you mean now, we do use this value here as a prop https://github.com/vitessce/vitessce/blob/e2ce2e4310b4ead6609c3611158add3b6951ddfd/packages/view-types/heatmap/src/HeatmapSubscriber.js#L312 so please ignore my suggestion above.

I guess this is why typescript is helpful. Can we leave a comment about this in the code here?